apps/src/aichat/redux/aichatRedux.ts

```diff
@@ -443,6 +447,46 @@ export const submitChatContents = createAsyncThunk(
   }
 );
 
+// This thunk's callback function submits a teacher's student's id along with the level/script id
+// (and the scriptLevelId if the level is a sublevel) to the student chat history endpoint,
+// waits for a response, and then returns the student's chat events for that level/script.
+export const fetchStudentChatHistory = createAsyncThunk(
+  'aichat/fetchStudentChatHistory',
+  async (studentUserId: number, thunkAPI) => {
+    const state = thunkAPI.getState() as RootState;
+    // Post teacher's student's user id to backend and retrieve student's chat history.
+    let studentChatHistoryApiResponse;
+    const currentLevel = getCurrentLevel(state);
+    // The scriptLevelId is sent to the backend if the current level is a sublevel so that we can
+    // correctly check if the teacher has permission to view the student's chat history.
+    const scriptLevelId = currentLevel.parentLevelId
+      ? getCurrentScriptLevelId(state)
+      : undefined;
+    try {
+      studentChatHistoryApiResponse = await getStudentChatHistory(
+        studentUserId,
+        parseInt(state.progress.currentLevelId || ''),
+        state.progress.scriptId,
+        scriptLevelId
+      );
+    } catch (error) {
+      Lab2Registry.getInstance()
+        .getMetricsReporter()
+        .logError(
+          'Error in aichat student chat history request',
+          error as Error
+        );
+      return;
+    }
+    const chatHistory: ChatEvent[] = studentChatHistoryApiResponse.map(
+      (event: string) => {
+        return JSON.parse(event);
+      }
+    );
+    thunkAPI.dispatch(setStudentChatHistory(chatHistory));
```

Comment:
I think this could be safely moved up inside the `try{}` block.

That way you could move the declaration of `studentChatHistoryApiResponse` inside the `try` block as well and make it a const.

Reply:
Oh nice! Will update.